tensorflow/contrib/seastar/seastar_worker_service.cc

```diff
@@ -0,0 +1,434 @@
+  Schedule([this, tag]() {
+      SeastarCall<TracingRequest, TracingResponse>
+        *call = new SeastarCall<TracingRequest, TracingResponse>();
+      InitSeastarServerTag(&call->req_, &call->resp_, tag);
+
+      Status s = worker_->Tracing(&call->req_, &call->resp_);
+      tag->ProcessDone(s);
+      delete call;
+  });
+}
+
+void SeastarWorkerService::RecvTensorHandlerRaw(SeastarServerTag *tag) {
+  // LOG(INFO) << "SeastarWorkerService::RecvTensorHandlerRaw";
+  Schedule([this, tag]() {
+      CallOptions* call_opts = new CallOptions;
+
+      SeastarCall<RecvTensorRequest, SeastarTensorResponse> *call =
+          new SeastarCall<RecvTensorRequest, SeastarTensorResponse>();
+
+      InitSeastarServerTag(&call->req_, &call->resp_, tag, [call] (const Status& s) {delete call;});
+
+      worker_->RecvTensorAsync(call_opts, &call->req_, &call->resp_,
+                               [tag, call, call_opts](const Status& s) {
+                                 delete call_opts;
+                                 tag->ProcessDone(s);
+                               });
+    });
+}
+
+void SeastarWorkerService::FuseRecvTensorHandlerRaw(SeastarServerTag *tag) {
+  // LOG(INFO) << "SeastarWorkerService::FuseRecvTensorHandlerRaw";
+  Schedule([this, tag]() {
+      CallOptions* call_opts = new CallOptions;
+
+      SeastarCall<FuseRecvTensorRequest, SeastarFuseTensorResponse> *call =
+          new SeastarCall<FuseRecvTensorRequest, SeastarFuseTensorResponse>();
+
+      InitSeastarServerTag(&call->req_, &call->resp_, tag, [call] (const Status& s) {delete call;});
+
+      worker_->FuseRecvTensorAsync(call_opts, &call->req_, &call->resp_,
+                                   [tag, call, call_opts](const Status& s) {
+                                     delete call_opts;
+                                     tag->ProcessDone(s);
+                                   });
+    });
+}
+
+void SeastarWorkerService::Schedule(std::function<void()> f) {
+  worker_->env()->compute_pool->Schedule(std::move(f));
+}
+
+WorkerEnv* SeastarWorker::env() {
+  return env_;
+}
+
+SeastarWorker::SeastarWorker(WorkerEnv* worker_env) : Worker(worker_env) {
+}
+
+void SeastarWorker::RecvTensorAsync(CallOptions* opts,
+                                    const RecvTensorRequest* request,
+                                    SeastarTensorResponse* response,
+                                    StatusCallback done) {
+    const int64 step_id = request->step_id();
+    const string& key = request->rendezvous_key();
+    Rendezvous::ParsedKey parsed;
+
+    Status s = Rendezvous::ParseKey(key, &parsed);
+    Device* src_dev = nullptr;
+    if (s.ok()) {
+      s = PrepareRecvTensor(parsed, &src_dev);
+    }
+    if (!s.ok()) {
+      LOG(WARNING) << "PrepareRecvTensor failed, tensor:" << key;
+      done(s);
+      abort();
+    }
+
+    // TODO(rangeng.llb): make call opts useful.
+    // opts->SetCancelCallback([this, step_id]() { AbortStep(step_id); });
```

Comment:
Any reason why you comment out these lines here?

Reply:
removed